opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectTransientDatasourcesResource.java

```diff
@@ -80,10 +96,66 @@ public Response createDatasource(Magma.DatasourceFactoryDto factoryDto) {
           .entity(Dtos.asDto(ds).build()).build();
     } catch(MagmaRuntimeException e) {
       MagmaEngine.get().removeTransientDatasource(uid);
+
+      if (e instanceof DatasourceParsingException) {
+        cacheDatarsourceParseErrorLog((DatasourceParsingException) e);
+      }
```

Comment:
no need to empty the cache since entries are evicted on timetolive or overwritten if a new entry is added  (for a given sessionid) before the eviction time